930-delegate-tasks/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNgHelper.java

```
@@ -240,17 +256,127 @@ private ServiceNowTaskNGResponse createTicketUsingServiceNowTemplate(
+  }
+
+  public ServiceNowTaskNGResponse updateMultipleTickets(
+      ServiceNowTaskNGParameters serviceNowTaskNGParameters, String ticketId, String ticketType) {
+    if (CHANGE_TASK.equalsIgnoreCase(serviceNowTaskNGParameters.getUpdateMultiple().getType())) {
+      ChangeTaskUpdateMultiple changeRequestUpdateMultiple =
+          (ChangeTaskUpdateMultiple) serviceNowTaskNGParameters.getUpdateMultiple().getSpec();
+
+      List<String> changeTaskIdsToUpdate =
+          fetchChangeTasksFromCR(ticketId, changeRequestUpdateMultiple.getChangeTaskType(), serviceNowTaskNGParameters);
+
+      Map<String, String> fieldsBody = serviceNowTaskNGParameters.getFields()
+                                           .entrySet()
+                                           .stream()
+                                           .filter(entry -> !StringUtils.isEmpty(entry.getValue()))
+                                           .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+      List<ServiceNowTicketNG> updatedTickets = new ArrayList<>();
+
+      List<Future<ServiceNowTaskNGResponse>> futures = new ArrayList<>();
+
+      log.info("Body of the update issue request made to the ServiceNow server: {}", fieldsBody);
+      for (String changeTaskId : changeTaskIdsToUpdate) {
+        Callable<ServiceNowTaskNGResponse> task =
+            () -> updateSingleTicket(serviceNowTaskNGParameters, changeTaskId, ticketType);
+
+        Future<ServiceNowTaskNGResponse> future = executorService.submit(task);
+        futures.add(future);
+      }
+
+      for (Future<ServiceNowTaskNGResponse> future : futures) {
+        try {
+          ServiceNowTaskNGResponse partialResult = future.get(TIME_OUT, TimeUnit.SECONDS);
+          updatedTickets.add(partialResult.getTicket());
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+          log.error("Failed to update ticket in updateMultipleTask", e);
+          throw new ServiceNowException(
```

We can throw the exception as here. 

If an ServiceNow exception is thrown while updating any of the tickets, will it be caught?